### DISCUSSION TITLE
How to validate that something is an array, not caring about the items in the array(performance)

### DISCUSSION THREAD

**Discussion by itsramiel** [open-circle/valibot#829] [Q&A] [answered]
Hi,

I would like to validate that something is an array without validating the items inside the array. Simple knowing that it is an array is enough for me. The return type could be: `Array<unknown>`, how can I do that?

I see that `array` always loops through its item, but for performance reasons I dont care to validate the items:
https://github.com/fabian-hiller/valibot/blob/d7ef5fbeac1236ed2e6d6b25f5ee52194d9a595f/library/src/schemas/array/array.ts#L91-L98

**Comment by itsramiel** [comment, marked as answer]
Like this maybe:
```ts
v.custom<Array<unknown>>((x) => Array.isArray(x))
```

or is there a better way?

**Reply by fabian-hiller** [reply]
Yes, this is the fastest approach. You could also write `v.array(v.unknown())` but its much slower compared.

**Reply by itsramiel** [reply]
Is there a use case os v.array(v.unknown)? Why cant v.array accept no arguments and then there would be no validation for the items and the type would be `Array<unknown>`?

It doesnt make sense to me to loop over the array and check the `unknown` schema since there is no such javascript concept in the beginning. What do you think of the suggestion above? Does it work well with other parts of the schemas?

**Reply by fabian-hiller** [reply]
Great idea! But on the other hand I think this is a very rare case and in most situations performance doesn't really matter here. If more people request this behavior, I will think about it.
